### ISSUE TITLE
Alveus Bingo Game (Monetization)

### ISSUE THREAD

**Issue by abdullahmorrison** [alveusgg/extension#86]
Since this concept has been tested on Maya's channel with her goat cam bingo and chat likes it, it makes sense to go into the extension too.

### Discussed in https://github.com/alveusgg/extension/discussions/83

<div type='discussions-op-text'>

<sup>Originally posted by **abdullahmorrison** May 21, 2023</sup>
It would be good to have an Alveus BINGO section that a user can access after paying 100 bits. This would only be available in the alveus sanctuary channel.

I made a bingo twitch extension for erobb221 and the chat loves it: https://dashboard.twitch.tv/extensions/f424a0vj5jic601up4k7iiplhd29ax-1.1.0

I made a bingo website a long time ago for Maya: https://mayahigabingo.netlify.app/
This bingo website is out of date but some tiles might still be relevant

Some tiles could also be. (These would maybe need approval from Maya or space)
- [ ] stompy bullies winnie
- [ ] winnie chewing cud
- [ ] reed pacing

Some viewer can help give more ideas for tiles</div>

**Comment by MattIPv4** [comment]
I think it'd be good to integrate this with the website, so that it talks to an API on the site to fetch the current bingo board. This'd allow us to configure custom bingo boards for specific streams, like the goat cam or a winnie cam.

**Comment by MattIPv4** [comment]
I'm less sure about the idea of locking it behind monetization, as I believe that'll need a permanently active server component to track who has paid, store that, and then allow the extension to check that on every load. That seems like a lot of overhead for a feature that'd be fun without it.

**Comment by abdullahmorrison** [comment]
> I think it'd be good to integrate this with the website, so that it talks to an API on the site to fetch the current bingo board. This'd allow us to configure custom bingo boards for specific streams, like the goat cam or a winnie cam.

this could work. I'm not interested in working on this but I can give you access to the code if needed. I made it in Angular though.

> I'm less sure about the idea of locking it behind monetization, as I believe that'll need a permanently active server component to track who has paid, store that, and then allow the extension to check that on every load. That seems like a lot of overhead for a feature that'd be fun without it.

I think this can be cached in localstorage. If so, I think that would be worth it since Alveus viewers are more likely to spend money. We could also paywall it for a month or 2 until the number of people paying slows down a lot and then make it free for everyone.

**Comment by MattIPv4** [comment]
> this could work. I'm not interested in working on this but I can give you access to the code if needed. I made it in Angular though.

Ah, I was thinking the Alveus website in this context, using the admin UI we already have in there.

> I think this can be cached in localstorage. If so, I think that would be worth it since Alveus viewers are more likely to spend money. We could also paywall it for a month or 2 until the number of people paying slows down a lot and then make it free for everyone.

Yah, potentially. Though for streams like goat cam today, I think we'd want it to be freely accessible to everyone, not behind a paywall.

**Comment by abdullahmorrison** [comment]
I agree, special bingo streams should be free for everyone. If it doesn't overcomplicate things, monetizing the normal alveus streams with bingo seems like a good move.

**Comment by pjeweb** [comment]
👀  I like the idea.

## Some ideas

How do you feel about something like wordle, where there is a new bingo every day, so viewers can compare? maybe with the option to play more (randomized) bingos?

How do viewers share their rng bingo win? I have seen the bingo extension, but not how the community interacts with it. Does it have a way to share their bingo on social media or in the chat? would it make sense to use our website backend to generate a short (hash based) URL to share with an image of their result (open graph enabled) or sth?

## Monetization

I feel kinda unsure about the legal aspects. I would want to check that this would be okay in terms of law, twitch TOS and 501c3 non-profit rules. I am note sure how deep you looked into this yet.

If we monetize it:
How about letting the user rng'ing the bingo until they like their layout, then make them pay the bits to enable it? That might make it less of a luck thing and more competitive.

## Implementation

I am happy to contribute, but would you want to take the lead on this, matt?

**Comment by pjeweb** [comment]
Oh and one more concern would be about handling the multiple channels. I think this can only be enabled for Alveus and Maya streams. We would need to make sure that this does not interfere with making the extension public for collabs. Might be easier after all to split it into multiple extensions?!

I am sure the channel owner will have to somehow give the extension permission to make payments, right?

**Comment by MattIPv4** [comment]
I think this'd be fine to build into the main extension (I'm not a fan of us having to maintain another codebase), but agree it would probably be gated to specific channels (maybe it also hits the site api to be enabled on collab channels too?).

**Comment by MattIPv4** [comment]
> I am happy to contribute, but would you want to take the lead on this, matt?

I would probably suggest you'd lead for now? I think the first step would actually be on the alveus site to build out the admin UI and API that'll be used to fetch the bingo board options? And then I'm happy to lead on the implementation in the extension, or you can do that too if you'd like, either works.

**Comment by MattIPv4** [comment]
(Fwiw I think this is probably a lower priority than getting notifications landed though)
